server/tools/build-keycloak.sh

```diff
@@ -87,14 +87,20 @@ rm -rf /opt/jboss/keycloak/standalone/configuration/standalone_xml_history
 bin/jboss-cli.sh --file=/opt/jboss/tools/cli/standalone-ha-configuration.cli
 rm -rf /opt/jboss/keycloak/standalone/configuration/standalone_xml_history
 
+###########
+# Garbage #
+###########
+
+rm -rf /opt/jboss/keycloak/standalone/tmp/auth
+rm -rf /opt/jboss/keycloak/domain/tmp/auth
+
 ###################
 # Set permissions #
 ###################
 
-echo "jboss:x:1000:jboss" >> /etc/group
-echo "jboss:x:1000:1000:JBoss user:/opt/jboss:/sbin/nologin" >> /etc/passwd
-chown -R jboss:jboss /opt/jboss
-chmod -R g+rw /opt/jboss
-
-rm -rf /opt/jboss/keycloak/standalone/tmp/auth
-rm -rf /opt/jboss/keycloak/domain/tmp/auth
+echo "jboss:x:0:root" >> /etc/group
+echo "jboss:x:1000:0:JBoss user:/opt/jboss:/sbin/nologin" >> /etc/passwd
+chown -R jboss:root /opt/jboss
+chmod -R 774 /opt/jboss
```

Comment:
This sets execute permissions on text files for owner and group. Why is this better than 'chmod -R g+rw,o+r /opt/jboss'?

Reply:
Actually, it would be better to just use 'chmod -R g=u /opt/jboss' so that the group has the same permissions as the user.

Reply:
You could use 'chmod g=u,o+r /opt/jboss' if you really think other needs read permissions. I would prefer 'chmod g=u,o= /opt/jboss'. If we agree that other doesn't need access then you can also remove the 'chmod -R +X /opt/jboss'.

Reply:
I've just tried your suggested approach:
```
echo "jboss:x:0:root" >> /etc/group
echo "jboss:x:1000:0:JBoss user:/opt/jboss:/sbin/nologin" >> /etc/passwd
chown -R jboss:root /opt/jboss
chmod g=u,o= /opt/jboss
```

Unfortunately the container crashes (the container is pushed to dockerhub so you can try it yourself):

```
$ docker run --user 123:0 docker.io/slaskawi/keycloak:KEYCLOAK-13020-3
(...)
Unable to read the logging configuration from 'file:/opt/jboss/keycloak/standalone/configuration/logging.properties' (java.io.FileNotFoundException: /opt/jboss/keycloak/standalone/configuration/logging.properties (Permission denied))
java.util.concurrent.ExecutionException: JBTHR00005: Operation failed
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.AsyncFutureTask.get(AsyncFutureTask.java:253)
	at org.jboss.as.server@10.0.0.Final//org.jboss.as.server.Main.main(Main.java:109)
	at org.jboss.modules.Module.run(Module.java:352)
	at org.jboss.modules.Module.run(Module.java:320)
	at org.jboss.modules.Main.main(Main.java:593)
Caused by: org.jboss.msc.service.StartException in service jboss.as: Failed to start service
```

In general, we can not assume that we will always be running the container as `jboss` (or `root`) user. OpenShift gives you a random user belonging to `root` group. I think, we should go one step further and allow running this container as a random user belonging to a random group. In other words, I'm slightly leaning towards https://github.com/keycloak/keycloak-containers/pull/258

Reply:
@douglaspalmer @slaskawi Though per [this comment](https://github.com/openshift/origin/issues/18974#issuecomment-374578684) it's possible to run the pod under the same UID in OpenShift too, not convinced we should follow the use of that.

But what about using the [recommended form](https://docs.openshift.com/container-platform/3.11/creating_images/guidelines.html#openshift-specific-guidelines), see `Support Arbitrary User IDs` in that section.

Would the
```
RUN chgrp -R 0 /some/directory && \
    chmod -R g+rwX /some/directory
```

form, suggested there, help?

Reply:
@slaskawi Confirmed the change works fine with both `OCP v4.3.5` & `OCP vv3.11.154`. But like mentioned earlier, I would suggest to change the `/opt/jboss` group ownership & mode to follow the official guide. IOW cherrypick [this commit](https://github.com/iankko/keycloak-1/commit/b717eb7d8f14faaa662436e2a5de2471206ecd08). That one works on (both versions too) & follows the guideline.

Reply:
I added suggested changes and tested them using this Docker image: `docker.io/slaskawi/keycloak:KEYCLOAK-13020-3`. Here is a short list of use cases I tested:

* [x] Installed Docker on Ubuntu Docker image and tested mapped user namespaces as suggested in [KEYCLOAK-13310](https://issues.redhat.com/browse/KEYCLOAK-13310)
* [x] Tested running as an arbitrary user `docker run --user 123 docker.io/slaskawi/keycloak:KEYCLOAK-13020-3`
* [x] Tested running as an arbitrary user belonging to an non-root group (this fails, but it's expected): `docker run --user 123:3 docker.io/slaskawi/keycloak:KEYCLOAK-13020-3`
* [ ] Testing on OpenShift 4 at the moment.